metals/src/main/scala/scala/meta/internal/decorations/SyntheticsDecorationProvider.scala

```diff
@@ -488,7 +488,8 @@ final class SyntheticsDecorationProvider(
         case param: m.Term.Param =>
           if (param.decltpe.isEmpty) List(param.name.pos.toSemanticdb) else Nil
         case cs: m.Case =>
-          explorePatterns(List(cs.pat)) ++ visit(cs.body)
+          // if the case is too long then it'll be too messy
+          visit(cs.body) // explorePatterns(List(cs.pat)) ++ visit(cs.body)
```

Comment:
I don't agree about removing this, since it might be useful to know the type in case. 

We might be over eager in case of inferred type for methods, we could for sure separate it under another flag (or rethink the settings here to be more useful).

Any other opininos?

Reply:
It could also be nice to provide more options like showInferredType = minimal| full | etc...

Reply:
+1 for more granularity